couchdb/server/compiler.py

```diff
@@ -172,3 +181,119 @@ def import_b64egg(b64str, egg_cache=None):
     finally:
         if egg_path is not None and os.path.exists(egg_path):
             os.unlink(egg_path)
+
+
+def require(ddoc, context=None, **options):
+    """Wraps design ``require`` function with access to design document.
+
+    :param ddoc: Design document.
+    :type ddoc: dict
+
+    :return: Require function object.
+
+    Require function extracts export statements from stored module within
+    design document. It could be used to access shared libraries of common used
+    functions, however it's available only for DDoc function set.
+
+    This function is from CommonJS world and works by detailed
+    `specification <http://wiki.commonjs.org/wiki/Modules/1.1.1>`_.
+
+    :param path: Path to stored module through document structure fields.
+    :type path: basestring
+
+    :param module: Current execution context. Normally, you wouldn't used this
+        argument.
+    :type module: dict
+
+    :return: Exported statements.
+    :rtype: dict
+
+    Example of stored module:
+        >>> class Validate(object):
+        >>>     def __init__(self, newdoc, olddoc, userctx):
+        >>>         self.newdoc = newdoc
+        >>>         self.olddoc = olddoc
+        >>>         self.userctx = userctx
+        >>>
+        >>>     def is_author():
+        >>>         return self.doc['author'] == self.userctx['name']
+        >>>
+        >>>     def is_admin():
+        >>>         return '_admin' in self.userctx['roles']
+        >>>
+        >>>     def unchanged(field):
+        >>>         assert (self.olddoc is not None
+        >>>                 and self.olddoc[field] == self.newdoc[field])
+        >>>
+        >>> exports['init'] = Validate
+
+    Example of usage:
+        >>> def validate_doc_update(newdoc, olddoc, userctx):
+        >>>     init_v = require('lib/validate')['init']
+        >>>     v = init_v(newdoc, olddoc, userctx)
+        >>>
+        >>>     if v.is_admin():
+        >>>         return True
+        >>>
+        >>>     v.unchanged('author')
+        >>>     v.unchanged('created_at')
+        >>>     return True
+
+    .. versionadded:: 0.11.0
+    .. versionchanged:: 1.1.0 Available for map functions.
+    """
+    context = context or DEFAULT_CONTEXT.copy()
```

Comment:
This patch will take care the case of "user really want a empty dictionary as `context`".
```diff
@@ -257,7 +257,7 @@ def require(ddoc, context=None, **options):
     .. versionadded:: 0.11.0
     .. versionchanged:: 1.1.0 Available for map functions.
     """
-    context = context or DEFAULT_CONTEXT.copy()
+    context = context if context is not None else DEFAULT_CONTEXT.copy()
     _visited_ids = []

     def require(path, module=None):
```